### ISSUE TITLE
12 hour time support in GUI

### ISSUE THREAD

**Issue by mdshw5** [victronenergy/venus#1095]
First off - thank you so much for supporting the DIY community! I really can't believe that these systems work so well. As an use from the USA, it does bother me that the system time seems to only report in 24h/military format:

<img width="480" alt="Screenshot 2023-06-03 at 9 00 42 PM" src="https://github.com/victronenergy/venus/assets/819554/5ccebc85-6ccc-4ad7-939f-34be6cc1cbc1">

However, with some community modifications we can customize the flow dashboard to display time in 12h format:

<img width="483" alt="Screenshot 2023-06-03 at 9 00 13 PM" src="https://github.com/victronenergy/venus/assets/819554/1b051011-c060-4875-b6e8-f39a87236ae2">

I would be grateful if you could consider a system-wide setting for time display format in some future version. Seems like many of your users in the USA might also appreciate this as well!

**Comment by jhofstee** [comment]
I am not that familiar with the am/pm thing. What would you expect entering a time should do?

**Comment by kwindrem** [comment]
In the US (and maybe only in the US), time is often shown as a 12 hour value with AM or PM appended to the end. Midnight is 12:00:00 AM,  noon is 12:00:00 PM.

In GuiMods, I use Qt.formatDateTime(new Date(), timeFormat) to return the text string either as 12 hour or 24 hour time.

timeFormat is "h:mm ap" for 12 hour time and "h:mm" for 24 hour time.

**Comment by mdshw5** [comment]
👆yes thanks @kwindrem for the succinct answer. It would be great (maybe only for US users) to have this time format display set globally in the GUI.

**Comment by thepaulcooper** [comment]
Some British people prefer am/pm but most, if not all, can easily switch from one to the other.

**Comment by jhofstee** [comment]
Yes, the displaying part is trivial, but what would you expect a UI to do when setting a time, use 24h format or a separate toggle for am/pm. That is less trivial since 13 p.m. should be disallowed etc.

**Comment by mdshw5** [comment]
I believe setting a time could still be done in 24 hour format. Most people should be able to deal with that.

**Comment by kwindrem** [comment]
Most people will never have to set the date and time (presence of an NTP server will do that automatically). In any case, I agree with mdshw5 that setting time in 24 hour format is fine.
